[Filters] Fix bug where changes to `appliedFilters` prop don't get respected (#10566)
Shopify/polaris: .changeset/three-trains-tie.md

---
'@shopify/polaris': minor
---

> Updated the Filters component to fix a bug where `appliedFilters` changes via prop were not being handled

Reviewer: ```suggestion
Fixed a bug in `Filters` where changes to the `appliedFilters` prop were not being handled
```

Author: One day I'll get one of these changeset messages correct 😅
